### PR TITLE
UIImage Scaling Fixes#3598

### PR DESCRIPTION
**Pull request by Vishwaak**
<!-- Thanks for submitting a pull request for Terasology! :-)
Please fill in some details about the PR, below.
If the PR contains source code please make sure to run Checkstyle on it first.
If you add unit tests we'll love you forever! 

### Contains

Added new method to handle the drawing image it will decide whether the image requires a Scale or a Fit. 

### How to test
Opening the game u will help u notice the change in the UI of the application.
The name of Terasology is zoomed out now and the preview of the world fits in perfectly.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

**Comment by Cervator** [comment]
This pull request **introduces 1 alert** when merging f3742fa6c603e959fefe10f6b3c5efff89313756 into bd335348b7bdbe4c2681682832ccb24028394349 - [view on LGTM.com](https://lgtm.com/projects/g/MovingBlocks/Terasology/rev/pr-01714183fbf17563c7a5d7e1cc1cb7750898c76f)

**new alerts:**

* 1 for Whitespace contradicts operator precedence

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by jellysnake** [comment]
Hey, thanks for the PR
I'm going to trigger the bot to test this PR

@GooeyHub: OK to test

Could you fix up that LGTM error? It's just complaining about the whitespace around the math operators.

Lastly, mentioning the issue so github links them,
#3598

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Vishwaak** [comment]
I corrected the error you mentioned about.

File: engine/src/main/java/org/terasology/rendering/nui/ScaleMode.java [hunk 1]
```diff
@@ -82,6 +82,29 @@ public Vector2f scaleForRegion(Rect2i region, int actualWidth, int actualHeight)
         public String toString() {
             return "tiled";
         }
+    },
+    Automatic_selector {
```
Comment (jellysnake) [inline engine/src/main/java/org/terasology/rendering/nui/ScaleMode.java]: This should be in ALL_CAPS to match the other values.

Comment (jellysnake) [inline engine/src/main/java/org/terasology/rendering/nui/ScaleMode.java]: This should be in ALL_CAPS to match the other values.

File: engine/src/main/java/org/terasology/rendering/nui/ScaleMode.java [hunk 2]
```diff
@@ -82,6 +82,29 @@ public Vector2f scaleForRegion(Rect2i region, int actualWidth, int actualHeight)
...
+        @Override
+        public Vector2f scaleForRegion(Rect2i region, int actualWidth, int actualHeight) {
+
+                if(actualHeight < 500)
```
Comment (jellysnake) [inline engine/src/main/java/org/terasology/rendering/nui/ScaleMode.java]: This really shouldn't work off of a hardcoded size. Why 500? It seems really arbitrary. Should we treat an image that's 700x700 the same as one that's 600x3?

File: engine/src/main/java/org/terasology/rendering/nui/ScaleMode.java [hunk 3]
```diff
@@ -82,6 +82,29 @@ public Vector2f scaleForRegion(Rect2i region, int actualWidth, int actualHeight)
...
+                {
+                    System.out.println("Hell of a job");
```
Comment (jellysnake) [inline engine/src/main/java/org/terasology/rendering/nui/ScaleMode.java]: Debug print statements should not be present in the final code.

Additionally the preferred method for printing is to use our provided logger. Using this logger looks like
```java
LoggerFactory.getLogger(<classNameHere>.class);
```
Standard practice is to place it into a static final field,
```java
private static final Logger logger = LoggerFactory.getLogger(<className>.class);
```

File: engine/src/main/java/org/terasology/rendering/nui/ScaleMode.java [hunk 4]
```diff
@@ -82,6 +82,29 @@ public Vector2f scaleForRegion(Rect2i region, int actualWidth, int actualHeight)
...
+                System.out.println(actualHeight);
+            float scale = Math.max((float) region.width() / actualWidth, (float) region.height() / actualHeight);
+            return new Vector2f(actualWidth * scale, (actualHeight * scale)-100);
+
+
+        }
+
+        @Override
+        public String toString() {
+            return "atomatic";
```
Comment (jellysnake) [inline engine/src/main/java/org/terasology/rendering/nui/ScaleMode.java]:
```suggestion
            return "automatic";
```

File: engine/src/main/java/org/terasology/rendering/nui/ScaleMode.java [hunk 5]
```diff
@@ -82,6 +82,29 @@ public Vector2f scaleForRegion(Rect2i region, int actualWidth, int actualHeight)
...
+        @Override
+        public Vector2f scaleForRegion(Rect2i region, int actualWidth, int actualHeight) {
+
+                if(actualHeight < 500)
+                {
+                    System.out.println("Hell of a job");
+                    float scale = Math.max((float) region.width() / actualWidth, (float) region.height() / actualHeight);
+                    return new Vector2f(actualWidth * scale, actualHeight * scale);
+                }
+                System.out.println(actualHeight);
```
Comment (jellysnake) [inline engine/src/main/java/org/terasology/rendering/nui/ScaleMode.java]: See my other comment about printing

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Uh oh, something went wrong with the build. Need to check on that

**Pull request closed without merging**